### PR TITLE
Add Gradle Enterprise instructions for maintainers.

### PR DESCRIPTION
**Pull request by cortinico**
Adding some instructions on how to use Gradle Enterprise for our maintainers.

### CODE REVIEW

**Comment by cortinico** [comment]
@3flex @marschwar @schalkms FYI ☝️

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -311,8 +311,30 @@ warning: Using the last argument as keyword parameters is deprecated (Ruby 2.7.0
...
+We do have access to a managed [Gradle Enterprise instance](6) that is publishing
+build scans for all the builds executed on CI (not from forks).
+
+This is extremely helpful to debug build failures and have access to remote build cache.
+Build scans are public so everyone can get insights on our build status.
+
+If you're a **maintaner** of a project under github.com/detekt/, you can request an access token
+to connect your local machine to the Gradle Enterprise instance, so you will also be publishign scans.
+
+To do you mush:
```
Comment (chao2zhang) [inline .github/CONTRIBUTING.md:325]:
```suggestion
You must follow the steps below:
```

**Review by chao2zhang — APPROVED** [review]

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -311,8 +311,30 @@ warning: Using the last argument as keyword parameters is deprecated (Ruby 2.7.0
...
+## Gradle Enterprise Access
+
+We do have access to a managed [Gradle Enterprise instance](6) that is publishing
+build scans for all the builds executed on CI (not from forks).
+
+This is extremely helpful to debug build failures and have access to remote build cache.
+Build scans are public so everyone can get insights on our build status.
+
+If you're a **maintaner** of a project under github.com/detekt/, you can request an access token
+to connect your local machine to the Gradle Enterprise instance, so you will also be publishign scans.
```
Comment (3flex) [inline .github/CONTRIBUTING.md:323]:
```suggestion
to connect your local machine to the Gradle Enterprise instance, so you will also be publishing scans.
```

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -311,8 +311,30 @@ warning: Using the last argument as keyword parameters is deprecated (Ruby 2.7.0
...
 
+## Gradle Enterprise Access
+
+We do have access to a managed [Gradle Enterprise instance](6) that is publishing
+build scans for all the builds executed on CI (not from forks).
+
+This is extremely helpful to debug build failures and have access to remote build cache.
+Build scans are public so everyone can get insights on our build status.
+
+If you're a **maintaner** of a project under github.com/detekt/, you can request an access token
```
Comment (3flex) [inline .github/CONTRIBUTING.md:322]:
```suggestion
If you're a **maintainer** of a project under github.com/detekt/, you can request an access token
```

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -311,8 +311,30 @@ warning: Using the last argument as keyword parameters is deprecated (Ruby 2.7.0
...
+to connect your local machine to the Gradle Enterprise instance, so you will also be publishign scans.
+
+You must follow the steps below:
+
+1. Email us at [info@detekt.dev][7] or get in touch with one of the existing maintainer.
```
Comment (3flex) [inline .github/CONTRIBUTING.md:327]:
```suggestion
1. Email us at [info@detekt.dev][7] or get in touch with one of the existing maintainers.
```

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -311,8 +311,30 @@ warning: Using the last argument as keyword parameters is deprecated (Ruby 2.7.0
...
+2. An account on https://ge.detekt.dev/ will be created for you, which you need to configure upon login (e.g. reset your password).
+3. Run the `./gradlew provisionGradleEnterpriseAccessKey` tasks from the Detekt root folder
```
Comment (3flex) [inline .github/CONTRIBUTING.md:329]:
```suggestion
3. Run the `./gradlew provisionGradleEnterpriseAccessKey` task from the detekt root folder
```

**Review by 3flex — CHANGES_REQUESTED** [review]
Few typos to correct, looks good though, I will try this soon!

There might also need to be a process to verify ownership of a GitHub account when an email request comes in? I'm not on the Slack channel or anywhere else so other than trusting me when I email through it might be a bit tricky as my email isn't published publicly on GitHub.

This won't happen often though so it's not a major deal.

**Comment by cortinico** [comment]
> There might also need to be a process to verify ownership of a GitHub account when an email request comes in? I'm not on the Slack channel or anywhere else so other than trusting me when I email through it might be a bit tricky as my email isn't published publicly on GitHub.
> This won't happen often though so it's not a major deal.

True. We might ask people to create a repo inside github.com/detekt with a name like `verify-...` or something (similary to what sonatype does).

As you said also, this won't happen that frequently.

**Review by 3flex — APPROVED** [review]

**Pull request merged**